Timeout for dirent cache

shyambits2004 opened the pull request:
Refresh attrs of a directory should invalidate the expired dirent cache.
Attribute timeout would be used as the expirty time.

Change-Id: Ib5436250c2a87c9384da4f036b16633ccd03a54b
Signed-off-by: root <root@qa102-89.qa.lab>

Commit:
Timeout for dirent cache

Refresh attrs of a directory should invalidate the expired dirent cache.
Attribute timeout would be used as the expirty time.

Change-Id: Ib5436250c2a87c9384da4f036b16633ccd03a54b
Signed-off-by: root <root@qa102-89.qa.lab>

satgs commented on src/include/cache_inode.h:
```diff
@@ -969,6 +973,32 @@ cache_inode_refresh_attrs(cache_entry_t *entry)
 		entry->obj_handle->attrs->acl = NULL;
 	}
 
+	if ((entry->type == DIRECTORY) &&
```
This can be done in trust_attrs function (like in the original code). If attrs are refreshed through some other path then cache_inode_is_attrs_valid may always return true not giving a chance to invalidate dentries. This should be taken care of.

shyambits2004 committed:
Expire dirent cache on attr expiry.

Invalidate the dirent cache after frequency. Current code might cause entry
to get missed due to mtime calculation.

Change-Id: Ib1d8dace7cc735e087873ea368564c72b728d385
Signed-off-by: shyam <shyambits2004@gmail.com>
